### PR TITLE
#163388112 users can bookmark articles

### PR DESCRIPTION
**Pull request by asirvex**
#### What does this PR do?
It enables a user to bookmark an article
#### Description of Task to be completed?

 - Create the bookmark functionality
 - List the bookmarked articles in the user profile

#### How should this be manually tested?
1. Clone the repo
```
git clone https://github.com/andela/ah-the-phoenix-frontend.git
```
2. Change it into the current directory
```
cd ah-the-phoenix-frontend
```
3. Checkout to the feature branch
```    
git checkout ft-bookmark-articles-163388112 
```
4. Install the required dependencies using 
```
npm install
```
5. Run the server
```
npm start
```
6. Login to the website then click the bookmark button on an article
7. you can check the bookmarked articles at the user profile

Any background context you want to provide?
N/A

What are the relevant pivotal tracker stories?
[#163388112](https://www.pivotaltracker.com/story/show/163388112)

Screenshots (if appropriate)

### CODE REVIEW

File: src/components/Articles/BookmarkArticle.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React, { Component } from 'react'
+import { bookmarkArticle } from "../../redux/actioncreators/bookmarkArticleActions";
+import { connect } from 'react-redux';
+import { Button } from 'semantic-ui-react';
+
+export class BookmarkArticle extends Component {
+    handleClick = () => {
+        this.props.bookmarkArticle("starting-up")
+        console.log("--laini--");
+    }
```
Comment (PaulinhoNdegwa) [inline src/components/Articles/BookmarkArticle.js]: What does this piece of code do? @asirvex

**Comment by PaulinhoNdegwa** [comment]
@asirvex You can rebase the changes in develop to reduce the number of files changed..

File: src/components/Articles/BookmarkArticle.js (hunk 1 above)
Comment (asirvex) [reply src/components/Articles/BookmarkArticle.js]: I have removed it. I was using it for testing the api requests

File: src/redux/actioncreators/unBookmarkArticleActions.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            return axiosWithToken.get('api/v1/favorites')
+        })
+        .then(res => {
+            dispatch(getBookmarkedSuccess(res.data.favorites))
+            dispatch(unBookmarkArticleSuccess())
+            toastr.success("Success!", "Article removed from the bookmarks");
+        })
+        .catch(error => {
+            dispatch(unBookmarkArticleFailure())
+            console.log(error);
```
Comment (dev-jey) [inline src/redux/actioncreators/unBookmarkArticleActions.js:28]: Remove `console.log` statement

**Pull request merged**